src/python/grpcio/grpc/_cython/_cygrpc/credentials.pxd.pxi

```diff
@@ -26,9 +26,9 @@ cdef int _get_metadata(
     grpc_credentials_plugin_metadata_cb cb, void *user_data,
     grpc_metadata creds_md[GRPC_METADATA_CREDENTIALS_PLUGIN_SYNC_MAX],
     size_t *num_creds_md, grpc_status_code *status,
-    const char **error_details) except * with gil
+    const char **error_details) except -1 with gil
 
-cdef void _destroy(void *state) except * with gil
+cdef void _destroy(void *state) noexcept with gil
```

Comment:
Looks like we're not using GIL for this function, we should change here to `nogil`.

Reply:
Done. Looks like it is defined as `nogil` in `credentials.pyx.pxi` but `with gil` in `credentials.pxd.pxi`. Don't know why there were no mismatch errors thrown during build